test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs

```diff
@@ -131,6 +131,26 @@ private static IServiceProvider BuildFallbackServiceProvider(IEnumerable<IServic
             return sc.BuildServiceProvider();
         }
 
+        public static string NormalizeExpectedUrl(this string url)
```

Comment:
Or better yet, a comparer. `Assert.Equal` allows for a `IEqualityComparer<>` parameter, so you could roll this into a `StringUrlComparer` and use that instead. That said, none of of our tests actually have query strings or url fragments - couldn't we just use `StringComparer.OrdinalIgnoreCase`